iwyu.cc

```diff
@@ -5318,20 +5314,36 @@ class IwyuAstConsumer
     if (CanIgnoreCurrentASTNode())
       return true;
 
+    bool can_fwd_decl = CanForwardDeclareType(current_ast_node());
     // If we're not in a forward-declare context, use of a template
     // specialization requires having the full type information.
-    if (!CanForwardDeclareType(current_ast_node()) || type->isTypeAlias()) {
+    if (!can_fwd_decl || type->isTypeAlias()) {
       const TemplateInstantiationData data = GetTplInstData(type);
       instantiated_template_visitor_.ScanInstantiatedType(
           current_ast_node(), data.resugar_map, data.provided_types);
     }
 
+    const TemplateDecl* tpl = type->getTemplateName().getAsTemplateDecl();
```

Comment:
Both of these vars are only used inside the `!InImplicitCode(...)` branch, is it possible to move them in? Or does that hit the 80-col limit and get noisy?

Reply:
I have simply overlooked that, thanks!